File: api-tsl/csharp/src/model/tsp/information/TSPInformationURI.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml;
using tr.gov.tubitak.uekae.esya.api.tsl.model.core;
using tr.gov.tubitak.uekae.esya.api.tsl.model.schemeinformation;
using tr.gov.tubitak.uekae.esya.api.tsl.util;

namespace tr.gov.tubitak.uekae.esya.api.tsl.model.tsp.information
{
    public class TSPInformationURI : BaseElement
    {
        private readonly IList<TSLURI> informationURIs = new List<TSLURI>(); 

        public TSPInformationURI(XmlDocument document, IList<TSLURI> iInformationURIs) : base(document)
        {
            addLineBreak();
            informationURIs = iInformationURIs;

            for (int i = 0; i < informationURIs.Count; i++)
            {
                mElement.AppendChild(informationURIs[i].Element);
                addLineBreak();
            }
        }

        public TSPInformationURI(XmlDocument document)
            : base(document)
        {
            addLineBreak();
        }

        public TSPInformationURI(XmlElement aElement)
            : base(aElement)
        {
            if(mElement.HasChildNodes)
            {
                XmlNodeList childList = aElement.ChildNodes;
                foreach (XmlNode child in childList)
                {
                    informationURIs.Add(new TSLURI((XmlElement)child));
                }
            }
            else
            {
                throw new TSLException(Constants.TAG_TSPINFORMATIONURI + TSL_DIL.NODE_CANNOT_BE_FOUND);
            }
        }

        public void RemoveAllInformationURI()
        {
            mElement.RemoveAll();
            informationURIs.Clear();
            addLineBreak();
        }

        public void addInformationURI(TSLURI tslURI)
        {
            informationURIs.Add(tslURI);
            mElement.AppendChild(tslURI.Element);
            addLineBreak();
        }

        public override string LocalName
        {
            get { return Constants.TAG_TSPINFORMATIONURI; }
        }

        public override string Namespace
        {
            get
            {
                return Constants.NS_TSL;
            }
        }

        public IList<TSLURI> InformationURIs
        {
            get { return informationURIs; }
        }

        public TSLURI InformationURIAt(int pos)
        {
            if(pos>=informationURIs.Count)
            {
                return null;
            }
            else
            {
                return informationURIs[pos];
            }
        }

        public string InformationURIStrAt(int pos)
        {
            if (pos >= informationURIs.Count)
            {
                return null;
            }
            else
            {
                return informationURIs[pos].TslURI;
            }
        }
    }
}

File: api-tsl/src/tr/gov/tubitak/uekae/esya/api/tsl/model/tsp/information/TSPInformationURI.java
package tr.gov.tubitak.uekae.esya.api.tsl.model.tsp.information;

import java.util.ArrayList;
import java.util.List;

import org.w3c.dom.Document;
import org.w3c.dom.Element;
import org.w3c.dom.Node;
import org.w3c.dom.NodeList;

import tr.gov.tubitak.uekae.esya.api.tsl.TSLException;
import tr.gov.tubitak.uekae.esya.api.tsl.model.core.BaseElement;
import tr.gov.tubitak.uekae.esya.api.tsl.model.schemeinformation.TSLURI;
import tr.gov.tubitak.uekae.esya.api.tsl.util.Constants;
import tr.gov.tubitak.uekae.esya.api.tsl.util.TSL_DIL;

public class TSPInformationURI extends BaseElement {
	private List<TSLURI> informationURIs = new ArrayList<TSLURI>();

	public TSPInformationURI(Document document, List<TSLURI> iInformationURIs) {
		super(document);
		addLineBreak();
		informationURIs = iInformationURIs;

		for (int i = 0; i < informationURIs.size(); i++) {
			mElement.appendChild(informationURIs.get(i).getElement());
			addLineBreak();
		}
	}

	public TSPInformationURI(Document document)

	{
		super(document);
		addLineBreak();
	}

	public TSPInformationURI(Element aElement) throws TSLException

	{
		super(aElement);
		if (mElement.hasChildNodes()) {
			NodeList childList = aElement.getElementsByTagName("*");
			for (int i = 0; i < childList.getLength(); i++) {
				Node child = childList.item(i);
				informationURIs.add(new TSLURI((Element) child));
			}
		} else {
			throw new TSLException(Constants.TAG_TSPINFORMATIONURI
					+ TSL_DIL.NODE_CANNOT_BE_FOUND);
		}
	}

	public void RemoveAllInformationURI() {
		RemoveAll();// mElement.RemoveAll();
		informationURIs.clear();
		addLineBreak();
	}

	public void addInformationURI(TSLURI tslURI) {
		informationURIs.add(tslURI);
		mElement.appendChild(tslURI.getElement());
		addLineBreak();
	}

	@Override
	public String getLocalName() {
		return Constants.TAG_TSPINFORMATIONURI;
	}

	@Override
	public String getNamespace() {
		return Constants.NS_TSL;

	}

	public List<TSLURI> getInformationURIs() {
		return informationURIs;
	}

	public TSLURI InformationURIAt(int pos) {
		if (pos >= informationURIs.size()) {
			return null;
		} else {
			return informationURIs.get(pos);
		}
	}

	public String InformationURIStrAt(int pos) {
		if (pos >= informationURIs.size()) {
			return null;
		} else {
			return informationURIs.get(pos).getTslURI();
		}
	}
}
